tests/cases/conformance/types/conditional/inferTypes1.ts

```diff
@@ -13,7 +13,7 @@ type T02 = Unpacked<() => string>;  // string
 type T03 = Unpacked<Promise<string>>;  // string
 type T04 = Unpacked<Unpacked<Promise<string>[]>>;  // string
 type T05 = Unpacked<any>;  // any
-type T06 = Unpacked<never>;  // never
+type T06 = Unpacked<never>;  // {}

```

Comment:
Isn't this counter-intuitive? o:

Reply:
@alfaproject `never extends T` is true for all T, so it just matches the first conditional, but fails to infer anything for `U`, resulting in `{}` instead of `never`. Based on playing around with some complex conditional types with @weswigham, I think this behaviour will result in *more* understandable types when composing complex conditionals.

Reply:
:+1:

Reply:
Gotcha, cheers (:

Reply:
Not sure what is more intuitive, but when inferring from type `never` to some type containing type parameters, I definitely think it is reasonable to infer `never` for those type parameters (same goes for the wildcard type that we use when determining whether to defer or resolve conditional types). I have added this propagation behavior to the PR. The effect is that we now infer `never` for `T06` above.